fix dynamodb index query validation

foxmulder900 opened the pull request:
Fixes an issue where the following exception was being thrown for a valid query:
```
One or more parameter values were invalid: Select type ALL_ATTRIBUTES is not supported for 
global secondary index id-index because its projection type is not ALL
```

The exception is thrown if the table has additional global secondary indexes (besides the one being queried) that do not have a projection type of ALL.

foxmulder900 commented on tests/integration/test_dynamodb.py:
```diff
@@ -258,6 +258,79 @@ def test_multiple_update_expressions(self, dynamodb):
+            )
+        assert ctx.match("ValidationException")
+
+    def test_valid_query_index(self, dynamodb):
```
This test fails without the code changes above.

foxmulder900 commented on localstack/services/dynamodb/dynamodb_listener.py:
```diff
@@ -1020,12 +1021,21 @@ def update_global_table(data):
     for index in schema["Table"].get("GlobalSecondaryIndexes", []):
-        index_projection_type = index.get("Projection").get("ProjectionType")
-        if index_query_type == "ALL_ATTRIBUTES" and index_projection_type != "ALL":
-            return False
```
This was returning False if _any_ index did not have a projection type of ALL, but only the index actually being queried should be checked.

whummer commented on tests/integration/test_dynamodb.py:
```diff
@@ -258,6 +258,84 @@ def test_multiple_update_expressions(self, dynamodb):
+                KeyConditionExpression=Key("field_a").eq("xyz"),
+                Select="ALL_ATTRIBUTES",
+            )
+        assert ctx.match("ValidationException")
```
nit: Let's better clean up and delete the table here at the end of the test method. (same in line 337 below)

whummer commented on localstack/services/dynamodb/dynamodb_listener.py:
```diff
@@ -1020,12 +1021,21 @@ def update_global_table(data):
+    table_name = to_str(query_data["TableName"])
+    index_name = to_str(query_data["IndexName"])
+    index_query_type = query_data.get("Select")
+    index = get_global_secondary_index(table_name, index_name)
```
Technically the `index` could be `None` here - can we raise an `Exception` and add a TODO comment: 

```
if not index:
    # TODO: add proper exception handling
    raise Exception("Index not found")
```

(we have another refactoring PR #5796 pending where we can then later pick this up.. 👍 ) Thanks

whummer reviewed (commented):
Great catch, thanks for this PR @foxmulder900  ! 👍 LGTM, only two minor comments - can we please address them before merging? Thanks

foxmulder900 commented:
@whummer Thanks for the review! (and all the work you do on LocalStack!)

I have addressed your comments, let me know if there is anything else to take care of.

whummer reviewed (approved):
LGTM, thanks for updating the PR @foxmulder900 ! 👍
